From 20e940bcfc1eb15f5b99255e9550b4bf63ce2a57 Mon Sep 17 00:00:00 2001
From: Jingyu Zhou <b02241@freescale.com>
Date: Tue, 12 Aug 2008 12:49:54 +0800
Subject: [PATCH] ENGR00087454 Add power init state

Add power init state

Signed-off-by: Zhou Jingyu <Jingyu.Zhou@freescale.com>
(cherry picked from commit 58ed49559b8d08596ef594c5b10bbdabd3f6c4b7)
---
 drivers/regulator/max8660/reg-max8660.c |    4 ++++
 include/asm-arm/arch-mxc/pmic_battery.h |   29 +++++++++++++++--------------
 2 files changed, 19 insertions(+), 14 deletions(-)

diff --git a/drivers/regulator/max8660/reg-max8660.c b/drivers/regulator/max8660/reg-max8660.c
index 63ed2eb..8da2494 100644
--- a/drivers/regulator/max8660/reg-max8660.c
+++ b/drivers/regulator/max8660/reg-max8660.c
@@ -624,6 +624,7 @@ static struct max8660_regulator reg_max8660[NUM_MAX8660_REGULATORS] = {
 		       .name = "SW1",
 		       .id = MCU_SW1,
 		       .ops = &max8660_sw1_ops,
+		       .use_count = 1,
 		       },
 	 },
 	{
@@ -631,6 +632,7 @@ static struct max8660_regulator reg_max8660[NUM_MAX8660_REGULATORS] = {
 		       .name = "SW2",
 		       .id = MCU_SW2,
 		       .ops = &max8660_sw2_ops,
+		       .use_count = 1,
 		       },
 	 },
 	{
@@ -639,6 +641,7 @@ static struct max8660_regulator reg_max8660[NUM_MAX8660_REGULATORS] = {
 		       .id = MCU_SW3,
 		       .ops = &max8660_sw3_ops,
 		       .constraints = &max8660_sw3_regulation_constraints,
+		       .use_count = 1,
 		       },
 	 },
 	{
@@ -647,6 +650,7 @@ static struct max8660_regulator reg_max8660[NUM_MAX8660_REGULATORS] = {
 		       .id = MCU_SW4,
 		       .ops = &max8660_sw4_ops,
 		       .constraints = &max8660_sw4_regulation_constraints,
+		       .use_count = 1,
 		       },
 	 },
 	{
diff --git a/include/asm-arm/arch-mxc/pmic_battery.h b/include/asm-arm/arch-mxc/pmic_battery.h
index 61dbdfc..e216436 100644
--- a/include/asm-arm/arch-mxc/pmic_battery.h
+++ b/include/asm-arm/arch-mxc/pmic_battery.h
@@ -180,20 +180,6 @@ typedef enum {
 	BAT_IT_BELOW_THRESHOLD,
 } t_batt_event;
 
-/*START: for 3ds hw event*/
-/*!
- * Battery event type enum
- */
-enum {
-	BAT_EVENT_CHARGER_PLUG = 0x01,
-	BAT_EVENT_CHARGER_UNPLUG = 0x02,
-	BAT_EVENT_CHARGER_OVERVOLTAGE = 0x04,
-	BAT_EVENT_BATTERY_LOW = 0x08,
-	BAT_EVENT_POWER_FAILED = 0x10,
-	BAT_EVENT_CHARGER_FULL = 0x20,
-} t_bat_event;
-/*END: for 3ds hw event*/
-
 /*!
  * This structure is used for the following battery changer control
  * IOCTLs:
@@ -239,8 +225,23 @@ typedef struct {
 } t_eol_setting;
 
 /* EXPORTED FUNCTIONS */
+
 #ifdef __KERNEL__
 
+/*START: for 3ds hw event*/
+/*!
+ * Battery event type enum
+ */
+enum {
+	BAT_EVENT_CHARGER_PLUG = 0x01,
+	BAT_EVENT_CHARGER_UNPLUG = 0x02,
+	BAT_EVENT_CHARGER_OVERVOLTAGE = 0x04,
+	BAT_EVENT_BATTERY_LOW = 0x08,
+	BAT_EVENT_POWER_FAILED = 0x10,
+	BAT_EVENT_CHARGER_FULL = 0x20,
+} t_bat_event;
+/*END: for 3ds hw event*/
+
 /*!
  * This function is used to start charging a battery. For different charger,
  * different c_voltage and current range are supported. \n
-- 
1.5.4.4

